java/src/main/native/OrtJniUtil.c

```diff
@@ -600,427 +588,473 @@ size_t copyPrimitiveArrayToJava(JNIEnv *jniEnv, ONNXTensorElementDataType onnxTy
+  if (code != ORT_OK) {
+    return NULL;
+  }
+  size_t * offsets = NULL;
+  code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(size_t), (void**)&offsets));
+  if (code == ORT_OK) {
     // Get a view on the String data
-    checkOrtStatus(jniEnv,api,api->GetStringTensorContent(tensor,characterBuffer,totalStringLength,offsets,1));
+    code = checkOrtStatus(jniEnv, api, api->GetStringTensorContent(tensor, characterBuffer, totalStringLength, offsets, 1));
 
-    size_t curSize = (offsets[0]) + 1;
-    characterBuffer[curSize-1] = '\0';
-    jobject tempString = (*jniEnv)->NewStringUTF(jniEnv,characterBuffer);
+    if (code == ORT_OK) {
+      size_t curSize = (offsets[0]) + 1;
+      characterBuffer[curSize-1] = '\0';
+      tempString = (*jniEnv)->NewStringUTF(jniEnv, characterBuffer);
+    }
 
-    checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,characterBuffer));
-    checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,offsets));
+    code = checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator, characterBuffer));
+    if (code != ORT_OK) {
+      tempString = NULL;
+    }
+  }
 
-    return tempString;
+  code = checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator, offsets));
+  if (code != ORT_OK) {
+    tempString = NULL;
+  }
+
+  return tempString;
 }
 
-void copyStringTensorToArray(JNIEnv *jniEnv, const OrtApi * api, OrtAllocator* allocator, OrtValue* tensor, size_t length, jobjectArray outputArray) {
+OrtErrorCode copyStringTensorToArray(JNIEnv *jniEnv, const OrtApi * api, OrtAllocator* allocator, OrtValue* tensor, size_t length, jobjectArray outputArray) {
+    char * tempBuffer = NULL;
     // Get the buffer size needed
-    size_t totalStringLength;
-    checkOrtStatus(jniEnv,api,api->GetStringTensorDataLength(tensor,&totalStringLength));
+    size_t totalStringLength = 0;
+    OrtErrorCode code = checkOrtStatus(jniEnv, api, api->GetStringTensorDataLength(tensor, &totalStringLength));
+    if (code != ORT_OK) {
+        return code;
+    }
 
     // Create the character and offset buffers
-    char * characterBuffer;
-    checkOrtStatus(jniEnv,api,api->AllocatorAlloc(allocator,sizeof(char)*(totalStringLength+length),(void**)&characterBuffer));
+    char * characterBuffer = NULL;
+    code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(char)*(totalStringLength+length), (void**)&characterBuffer));
+    if (code != ORT_OK) {
+        return code;
+    }
     // length + 1 as we need to write out the final offset
-    size_t * offsets;
-    checkOrtStatus(jniEnv,api,api->AllocatorAlloc(allocator,sizeof(size_t)*(length+1),(void**)&offsets));
+    size_t * offsets = NULL;
+    code = checkOrtStatus(jniEnv, api, api->AllocatorAlloc(allocator, sizeof(size_t)*(length+1), (void**)&offsets));
+    if (code != ORT_OK) {
+        checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,characterBuffer));
```

Comment:
>checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,characterBuffer)); [](http://example.com/codeflow?start=8&length=73)

This swallows the allocation error exception which is more interesting than potential deallocation error.

Reply:
Fixed.